internal/internal_worker_base.go

```diff
@@ -167,11 +168,18 @@ func createPollRetryPolicy() backoff.RetryPolicy {
 func newBaseWorker(options baseWorkerOptions, logger *zap.Logger, metricsScope tally.Scope, sessionTokenBucket *sessionTokenBucket) *baseWorker {
 	ctx, cancel := context.WithCancel(context.Background())
 
+	dynamic := &worker.DynamicParams{
+		PollerPermit: worker.NewPermit(options.pollerCount),
+		TaskPermit:   worker.NewPermit(options.maxConcurrentTask),
+	}
+
 	var pollerAS *pollerAutoScaler
 	if pollerOptions := options.pollerAutoScaler; pollerOptions.Enabled {
+		dynamic.PollerPermit = worker.NewPermit(pollerOptions.InitCount)
```

Comment:
why not use the same `options.pollerCount` to initialize PollerPermit when autoscale is enabled?

Reply:
pollerCount is actually the maximum number of pollers. Without poller auto scaler, this is the "initial" count. For poller auto scaler, I was trying to override the initial poller count by not throttling poller requests when the worker first starts up.